### ISSUE TITLE
Autocorrect of Lint/Loop can be unsafe

### ISSUE THREAD

**Issue by jurriaan** [rubocop-hq/rubocop#9011]
##  Expected behavior

Safe auto correct of Lint/Loop should not introduce errors in the code

## Actual behavior

A `begin`...`end` `until` loop introducing a local variable gets corrected to a `loop do` .. `break if` .. `end` structure which has its own local variable scope, potentially breaking the code

## Steps to reproduce the problem

The following code:

```ruby
def some_check?
  true
end

def generate_key
  42
end

begin
  key = generate_key
end until some_check?

puts key
```

gets auto corrected (run `rubocop -a` on the file) by the `Lint/Loop` cop to the following:

```ruby
def some_check?
  true
end

def generate_key
  42
end

loop do
  key = generate_key
  break if some_check?
end

puts key
```

which generates the following error instead of printing `42`:

```
Traceback (most recent call last):
test.rb:14:in `<main>': undefined local variable or method `key' for main:Object (NameError)
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.2.0 (using Parser 2.7.2.0, rubocop-ast 1.1.0, running on ruby 2.7.2 x86_64-linux)
```
